### DISCUSSION TITLE
How to merge secrets into Helm values files properly?

### DISCUSSION THREAD

**Discussion by ntjn** [fluxcd/flux2#2227] [Q&A] [answered]
Assume a simple Secret object is stored in a K8S cluster.

I want to merge 2 fields from it to my values file with a values file specified for a chart for a given environment.
Based on the docs there are multiple solutions for that:

1. Use the `valuesFrom` directive \

    * However, I also use `valuesFiles`, and `valuesFrom` does not seem to override selected fields.
    * Maybe because `valuesFiles` has priority?

2. Use Kustomization's [configMapGenerator](https://fluxcd.io/docs/guides/helmreleases/#refer-to-values-in-configmaps-generated-with-kustomize) and merge this with Secrets

     Kustomization appends an id to the created configMap, so it can't be referred, only if specified by another commit if the value file changes.

Am I right?
Is there any other other solution for my problem?

**Comment by hiddeco** [comment, marked as answer]
`valuesFrom` can overwrite a specific field if you define a `targetPath`: https://fluxcd.io/docs/guides/helmreleases/#refer-to-values-in-configmap-and-secret-resources

**Reply by ntjn** [reply]
Yes, it really works. (There was a bug in my chart.) Thanks for the quick reply.
